### PR TITLE
build: add semantic-release for automatic package publishing

### PR DESCRIPTION
**Pull request by nodkz**
@jbdemonte needs to setup proper access tokens between CI, Git and Npm. My current accesses do not allow to generate tokens. And no need on regular bases, cause it needs to do only once.

It can be done by you in couple of seconds via https://github.com/semantic-release/semantic-release/blob/caribou/docs/usage/ci-configuration.md#automatic-setup-with-semantic-release-cli

Needs to globally install  `semantic-release-cli`.
Run `semantic-release-cli setup`
Answer on several questions. When it asks about the creation of Travis.yml just skip, cause it already exists and setup correctly.
Under the hood it connects to Travis, Git and Npm and creates proper tokens for each other.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "mongoose-elasticsearch-xp",
-  "version": "5.2.0",
+  "version": "0.0.0-semantically-released",
```
Comment (jbdemonte) [inline package.json]: should not it be updated to 5.2.0 ?

Comment (nodkz) [reply package.json]: Nope. It should be in such form.

Semantic-release calculates new version such way:
- take last published version from npm
- try to find a `tag` in github repo with such version
- take commit messages after founded `tag`
- look at commit prefixes and decide which version to publish major/minor/fix
- after that it get npm version from first step and add proper numbers

So `version` value from `package.json` does not use and should contain some informative notice. In our case `0.0.0-semantically-released`.

Comment (nodkz) [reply package.json]: Just run `semantic-release-cli setup` and i take care of rest problems if they occur. ;)

**Comment by nodkz** [comment]
Looking on build errors

**Pull request merged**